/hold
The test can be run multiple times
```
go test -run ^TestSyncLabels$ ./pkg/kubelet -count 10
ok      k8s.io/kubernetes/pkg/kubelet   3.068s
```

but it panics running with `stress`

```sh
$ go test -run ^TestSyncLabels$ ./pkg/kubelet -c
$ stress ./kubelet.test -test.run ^TestSyncLabels$

/tmp/go-stress-20211118T000810-591795816
E1118 00:08:11.101991 3904941 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 114 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3648b60, 0x6461020})
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x7d
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xfffffffe})
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x3648b60, 0x6461020})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
k8s.io/kubernetes/pkg/kubelet.TestSyncLabels.func1.2()
        /home/aojea/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_test.go:2429 +0x30e
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x410065, 0xc000b3de28})
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:220 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x43c7d28, 0xc00005c0c8}, 0x443860)
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:233 +0x7c
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x203000)
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:226 +0x39
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x5f5e100, 0x4008000000000000, 0x0, 0x6, 0x0}, 0x64b9400)
        /home/aojea/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:421 +0x5f
k8s.io/kubernetes/pkg/kubelet.retryWithExponentialBackOff(...)
        /home/aojea/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_test.go:2516
k8s.io/kubernetes/pkg/kubelet.TestSyncLabels.func1(0xc0002429c0)
        /home/aojea/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_test.go:2401 +0x25d
testing.tRunner(0xc0002429c0, 0xc00027df30)
        /usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1306 +0x35a
--- FAIL: TestSyncLabels (0.00s)
    --- FAIL: TestS
```

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -2410,9 +2410,12 @@ func TestSyncLabels(t *testing.T) {
 						}
 						for _, action := range actions {
 							if action.GetVerb() == "patch" {
-								patchAction := action.(core.PatchActionImpl)
-								var err error
-								savedNode, err = applyNodeStatusPatch(test.existingNode, patchAction.GetPatch())
+								var (
+									err          error
+									patchAction  = action.(core.PatchActionImpl)
+									patchContent = patchAction.GetPatch()
+								)
+								savedNode, err = applyNodeStatusPatch(test.existingNode, patchContent)
 								if err != nil {
 									t.Logf("node patching failed, %v", err)
 									return false, nil
@@ -2421,6 +2424,10 @@ func TestSyncLabels(t *testing.T) {
 						}
 					} else {
 						savedNode = test.existingNode
+					}
+					if savedNode == nil || savedNode.Labels == nil {
+						t.Logf("savedNode.Labels should not be nil")
+						return false, nil
 					}
 					val, ok := savedNode.Labels[v1.LabelOSStable]
 					if !ok {

